### PR TITLE
Add host-bulk insert_and_find for static_map

### PR DESCRIPTION
**Pull request by srinivasyadav18**
This PR adds **host-bulk** `insert_and_find` API for `static_map`.

- [x] Updates insert_and_find unit tests.
- [x] Adds documentation. 

Contributes to #477

### CODE REVIEW

**Comment by srinivasyadav18** [comment]
@PointKernel One of the test on **pre-volta arch** fails due to `static_assert`, as we don't support packed type (size > 8) writes on pre-volta architecture. I cannot disable those tests from hostside because `__CUDA_ARCH__` is not defined on hostside.  May I know how should be move further with this issue. Thanks!

**Comment by PointKernel** [comment]
> @PointKernel One of the test on **pre-volta arch** fails due to `static_assert`, as we don't support packed type (size > 8) writes on pre-volta architecture. I cannot disable those tests from hostside because `__CUDA_ARCH__` is not defined on hostside. May I know how should be move further with this issue. Thanks!

voila https://github.com/NVIDIA/cuCollections/blob/63d313eb015ab756f3ee7665af3b4b5b274e17fc/tests/static_map/custom_type_test.cu#L104-L109

File: tests/static_map/insert_and_find_test.cu [hunk 1]
```diff
@@ -23,57 +23,16 @@
 #include <thrust/device_vector.h>
 #include <thrust/execution_policy.h>
 #include <thrust/functional.h>
-#include <thrust/sequence.h>
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/iterator/transform_iterator.h>
+
+#include <cuda.h>
```
Comment (PointKernel) [inline tests/static_map/insert_and_find_test.cu]: do we need this header?

File: tests/static_map/insert_and_find_test.cu [hunk 2]
```diff
@@ -23,57 +23,16 @@
...
 
 #include <catch2/catch_template_test_macros.hpp>
 
-static constexpr int Iters = 10'000;
+#include <iostream>
```
Comment (PointKernel) [inline tests/static_map/insert_and_find_test.cu]:
```suggestion
```
could be removed

File: tests/static_map/insert_and_find_test.cu (hunk 1 above)
Comment (srinivasyadav18) [reply tests/static_map/insert_and_find_test.cu]: cuda.h is not requried. I missed this during cleanup.

**Review by PointKernel — APPROVED** [review]

**Review by sleeepyjack — APPROVED** [review]
Nice work!

**Pull request merged**